rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -1513,7 +1750,18 @@ def batch_loss(
     ) -> tf.Tensor:
         tf_batch_data = self.batch_to_model_data_format(batch_in, self.data_signature)
 
-        sequence_lengths = self._get_sequence_lengths(tf_batch_data[TEXT_SEQ_LENGTH][0])
+        mask_sequence_text = self._get_mask_for(tf_batch_data, TEXT_SEQUENCE_LENGTH)
+        mask_sentence_text = self._get_mask_for(tf_batch_data, TEXT_SENTENCE_LENGTH)
+
+        if TEXT_SEQUENCE_LENGTH in tf_batch_data:
+            sequence_lengths = self._get_sequence_lengths(
+                tf_batch_data[TEXT_SEQUENCE_LENGTH][0]
+            )
+            sequence_lengths += 1  # add cls token
+        else:
+            sequence_lengths = self._get_sequence_lengths(
+                tf_batch_data[TEXT_SENTENCE_LENGTH][0]
+            )
```

Comment:
should this simply be `1`, then we add `sequence_lengths` if `TEXT_SEQUENCE_LENGTH` exists

Reply:
same for labels

Reply:
unfortunately that does not work for response selector, setting the `sequence_lengths` to 1 works for DIET but for the Reponse Selector it causes a share mismatch `Dimensions of inputs should match: shape[0] = [1,1] vs. shape[1] = [64,1]`. So we need to know the batch dimension, but we cannot get that during `batch_loss`.

Reply:
who raises this error?

Reply:
solved it, was using `tensor.shape[0]`, which is not working in non-eager mode, switched to `tf.shape(tensor)[0]`.